rust/projects/file-io/project.md

```diff
@@ -64,17 +94,22 @@ methods:
   Return an error if the value is not read successfully.
 
 When setting a key to a value, `kvs` writes the `set` command to disk in a
-write-ahead log. On startup, the commands in the WAL are re-evaluated and the
-state cached in memory.
+sequential log. On startup, the commands in the log are re-evaluated and the log
+pointer (file offset) of the last command to modify each key recorded in the
+in-memory index.
+
+When retrieving a value for a key with the `get` command, it searches the index,
+and if found then loads from the log, and evaluates, the command at the
+corresponding log pointer.
 
-When the size of the WAL reaches a given threshold, `kvs` writes it to disk in a
-searchable format and clears the WAL. When retrieving a value for a key with the
-`get` command, first the in-memory WAL cache is searched, and if not found, then
-the on-disk values are searched.
+When the size of the uncompacted log entries reach a given threshold, `kvs`
+compacts it into a new log, removing redundent entries to reclaim disk space.
 
 Note that our `kvs` project is both a stateless command-line program, and a
-library containing a stateful `KvStore` type, and thus have different
-requirements.
+library containing a stateful `KvStore` type: for CLI use the `KvStore` type
+will load the index, execute the command, then exit; for library use it will
+load the index, then execute multiple commands, maintaining the index state,
+until it is dropped.
 
 
 ## Project setup
```

Comment:
Ask students to add necessary dev-dependencies so that tests can run.

Reply:
Done.